File: src/main/scala/resonantinduction/mechanical/mech/turbine/TurbineNode.java
package resonantinduction.mechanical.mech.turbine;

import resonantinduction.mechanical.mech.MechanicalNode;
import net.minecraftforge.common.util.ForgeDirection;

/**
 * Turbine's Mechanical node
 * Turbines always face forward and connect from behind.
 *
 * @author Calclavia, Darkguardsman
 */
public class TurbineNode extends MechanicalNode
{
	public TurbineNode(TileTurbine tileTurbineBase)
	{
		super(tileTurbineBase);
	}

	public TileTurbine turbine()
	{
		return (TileTurbine) getParent();
	}

	@Override
	public boolean canConnect(ForgeDirection from, Object source)
	{
		return turbine().getMultiBlock().isPrimary() && source instanceof MechanicalNode && !(source instanceof TurbineNode) && from == turbine().getDirection();
	}

	@Override
	public boolean inverseRotation(ForgeDirection dir)
	{
		return dir == turbine().getDirection().getOpposite();
	}
}

File: src/main/scala/resonantinduction/mechanical/mech/turbine/TurbineNode.scala
package resonantinduction.mechanical.mech.turbine

import net.minecraftforge.common.util.ForgeDirection
import resonantinduction.mechanical.mech.grid.NodeMechanical

/**
 * Turbine's Mechanical node
 * Turbines always face forward and connect from behind.
 *
 * @author Calclavia, Darkguardsman
 */
class TurbineNode(parent: TileTurbine) extends NodeMechanical(parent)
{

  /**
   * Moment of inertia = m * r * r
   * Where "m" is the mass and "r" is the radius of the object.
   */
  override def momentOfInertia = parent.multiBlockRadius * parent.multiBlockRadius

  def turbine: TileTurbine =
  {
    return getParent.asInstanceOf[TileTurbine]
  }

  override def canConnect[B](other: B, from: ForgeDirection): Boolean =
  {
    return turbine.getMultiBlock.isPrimary && other.isInstanceOf[NodeMechanical] && !(other.isInstanceOf[TurbineNode]) && from == turbine.getDirection
  }

  /**
  override def inverseRotation(dir: ForgeDirection): Boolean =
  {
    return dir == turbine.getDirection.getOpposite
  } */
}